ui/tests/integration/components/json-editor-test.js

```diff
@@ -122,12 +123,27 @@ module('Integration | Component | json-editor', function (hooks) {
     assert.dom('.CodeMirror-code').hasText(`{ "test": "********"}`, 'shows data with obscured values');
     assert.dom('[data-test-toggle-input="revealValues"]').isNotChecked('reveal values toggle is unchecked');
     await click('[data-test-toggle-input="revealValues"]');
-    assert.dom('.CodeMirror-code').hasText(JSON_BLOB, 'shows data with real values');
+    assert.strictEqual(codemirror().getValue(), JSON_BLOB, 'shows data with real values');
     assert.dom('[data-test-toggle-input="revealValues"]').isChecked('reveal values toggle is checked');
     // turn obscure back on to ensure readonly overrides reveal setting
     await click('[data-test-toggle-input="revealValues"]');
     this.set('readOnly', false);
     assert.dom('[data-test-toggle-input="revealValues"]').doesNotExist('reveal values toggle is hidden');
-    assert.dom('.CodeMirror-code').hasText(JSON_BLOB, 'shows data with real values on edit mode');
+    assert.strictEqual(codemirror().getValue(), JSON_BLOB, 'shows data with real values on edit mode');
+  });
+
+  test('code-mirror modifier sets value correctly on non json object', async function (assert) {
+    this.value = '#A comment';
+    await render(hbs`
+      <JsonEditor
+        @value={{this.value}}
+        @example={{this.example}}
+        @mode="ruby"
+        @valueUpdated={{fn (mut this.value)}}
+      />
+    `);
+    assert.strictEqual(codemirror().getValue(), `#A comment`, 'shows initial non json value');
+    codemirror().setValue('#Another comment');
```

Comment:
confirmed that the conditional was iterated through and properly handled a non-json situation.

Reply:
by using `setValue()` are we bypassing the codemirror functions entirely? I think what we want to test here is that using `fillIn` or `typeIn` - which more accurately mocks user interaction - doesn't bork the editor which eventually calls `setValue()` Does this make sense?